File: pubsub/src/main/java/glitch/pubsub/events/ClearChatEvent.java
package glitch.pubsub.events;

import glitch.api.ws.events.IEvent;
import glitch.pubsub.GlitchPubSub;
import glitch.pubsub.Topic;
import glitch.pubsub.object.json.Moderator;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class ClearChatEvent extends AbstractMessageEvent<Moderator> implements IEvent<GlitchPubSub> {
    public ClearChatEvent(GlitchPubSub client, Topic topic, Moderator data) {
        super(client, topic, data);
    }
}

File: pubsub/src/main/kotlin/glitch/pubsub/events/ClearChatEvent.kt
package glitch.pubsub.events

import glitch.api.ws.events.IEvent
import glitch.pubsub.GlitchPubSub
import glitch.pubsub.Topic
import glitch.pubsub.`object`.json.Moderator

class ClearChatEvent(client: GlitchPubSub, topic: Topic, data: Moderator) : AbstractMessageEvent<Moderator>(client, topic, data), IEvent<GlitchPubSub>
